src/Model.jl

```diff
@@ -95,6 +99,7 @@ function Model(lattice::AbstractMatrix{T};
     else
         @assert n_electrons isa Int
     end
+    isempty(terms) && error("Model without terms not supported.")
```

Comment:
Can't you just leave this as a warning? As the rest of this PR shows it's sometimes useful for tests

Reply:
... which is why the default is now to make it a "Kinetic only" if you specify nothing. I have to say I find "model without terms" rather odd and I could imagine that we have code that implicitly assumes this (I have not checked). Certainly it is an easy assumption you might make without thinking about it, because to be fair I cannot imagine a realistic use case where you want a Model with truly no term in it.

But sure I can make it a warning if you insist.

Reply:
I'm fine with whatever, it just looked simpler to allow it for tests but it's fine this way too!